frontbackend/backend.py

```diff
@@ -0,0 +1,127 @@
+"""
+Front (front.com) email backend.
+
+There's some differences from normal email:
+
+- There are a few settings:
+  * FRONT_TOKEN - the API token used in the Authorization header
+  * FRONT_CHANNEL - the Front inbox to use
+  * FRONT_AUTHOR - the author for a draft message; ONLY used with drafts
+  * FRONT_ARCHIVE (default=False) - whether to archive sent messages
+- "From" and "Reply to" aren't used.
+  Instead, the from address is set by the channel (FRONT_CHANNEL).
+- By appending an attribute "draft" to an email message,
+  the backend will create a draft instead of sending the message.
+"""
+import logging
+
+import requests
+from django.core.mail.backends.base import BaseEmailBackend
+from django.core.mail.message import sanitize_address
+
+from . import settings
+
+
+log = logging.getLogger(__name__)  # noqa
+
+
+class EmailBackend(BaseEmailBackend):
+
+    """
+    Custom email backend to send messages through Front (front.com).
+
+    See: https://docs.djangoproject.com/en/3.2/topics/email/#defining-a-custom-email-backend
+    """
+
+    def __init__(self, **kwargs):
+        """Init override."""
+        super().__init__(**kwargs)
+
+        token = settings.FRONT_TOKEN
+        channel = settings.FRONT_CHANNEL
+
+        if not token or not channel:
+            raise NotImplementedError(
+                "For the Front email backend, "
+                "settings.FRONT_TOKEN and settings.FRONT_CHANNEL "
+                "must be set."
+            )
+
+        # Whether to archive messages on send
+        self.archive = settings.FRONT_ARCHIVE
```

Comment:
Seems like we should be able to set this per message, right? The current implementation allows the user to choose whether to archive it which is useful if you want to follow up on it. Probably not super important though.

 https://github.com/readthedocs/ethical-ad-server/blob/192f6c6af45a740f140f65e9d308c88b04897627/adserver/staff/forms.py#L352

Reply:
I guess it's easy enough to make this configure the default and allow some way to override it.